x-pack/legacy/plugins/maps/public/layers/styles/components/vector/color/dynamic_color_selection.js

```diff
@@ -17,15 +17,17 @@ export function DynamicColorSelection({ ordinalFields, onChange, styleOptions })
     onChange({ ...styleOptions, field });
```

Comment:
I've been getting some type-validation errors in the console.

This one triggers when selecting the custom option and the dynamic field has not been selected yet
``` .
Warning: Failed prop type: Invalid prop `fillColor.options` supplied to `VectorIcon`.
    in VectorIcon (created by LayerTocActions)
    in LayerTocActions (created by TOCEntry)
    in div (created by TOCEntry)
    in div (created by TOCEntry)
    in TOCEntry (created by Connect(TOCEntry))
    in Connect(TOCEntry) (created by DragHandle)
    in div (created by DragHandle)
    in DragHandle (created by Draggable)
    in DraggableDimensionPublisher (created by Draggable)
    in Draggable (created by Connect(Draggable))
    in Connect(Draggable) (created by EuiDraggable)
    in EuiDraggable (created by Droppable)
    in div (created by Droppable)
    in DroppableDimensionPublisher (created by Droppable)
    in Droppable (created by Connect(Droppable))
    in Connect(Droppable) (created by EuiDroppable)
    in EuiDroppable (created by LayerTOC)
    in ErrorBoundary (created by DragDropContext)
    in DragDropContext (created by EuiDragDropContext)
    in EuiDragDropContext (created by LayerTOC)
    in div (created by LayerTOC)
    in LayerTOC (created by Connect(LayerTOC))
    in Connect(LayerTOC) (created by LayerControl)
    in div (created by EuiFlexItem)
    in EuiFlexItem (created by LayerControl)
    in div (created by EuiPanel)
    in EuiPanel (created by LayerControl)
    in LayerControl (created by Connect(LayerControl))
    in Connect(LayerControl) (created by WidgetOverlay)
    in div (created by EuiFlexItem)
    in EuiFlexItem (created by WidgetOverlay)
    in div (created by EuiFlexGroup)
    in EuiFlexGroup (created by WidgetOverlay)
    in WidgetOverlay (created by Connect(WidgetOverlay))
    in Connect(WidgetOverlay) (created by GisMap)
    in div (created by EuiFlexItem)
    in EuiFlexItem (created by GisMap)
    in div (created by EuiFlexGroup)
    in EuiFlexGroup (created by GisMap)
    in GisMap (created by Connect(GisMap))
    in Connect(GisMap)
    in IntlProvider (created by I18nProvider)
    in I18nProvider
    in Provider
```

Not sure how I triggered these ones:
```
checkPropTypes.js:20 Warning: Failed prop type: The prop `styleOptions.color` is marked as required in `DynamicColorSelection`, but its value is `null`.
    in DynamicColorSelection (created by StaticDynamicStyleRow)
    in StaticDynamicStyleRow (created by VectorStyleColorEditor)
    in VectorStyleColorEditor (created by VectorStyleEditor)
    in VectorStyleEditor (created by StyleSettings)
    in div (created by EuiPanel)
    in EuiPanel (created by StyleSettings)
    in StyleSettings (created by Connect(StyleSettings))
    in Connect(StyleSettings) (created by LayerPanel)
    in div (created by LayerPanel)
    in div (created by LayerPanel)
    in div (created by EuiFlexGroup)
    in EuiFlexGroup (created by LayerPanel)
    in LayerPanel (created by Connect(LayerPanel))
    in Connect(LayerPanel) (created by GisMap)
    in div (created by EuiFlexItem)
    in EuiFlexItem (created by GisMap)
    in div (created by EuiFlexGroup)
    in EuiFlexGroup (created by GisMap)
    in GisMap (created by Connect(GisMap))
    in Connect(GisMap)
    in IntlProvider (created by I18nProvider)
    in I18nProvider
    in Provider
printWarning @ checkPropTypes.js:20
checkPropTypes @ checkPropTypes.js:82
validatePropTypes @ react.development.js:1666
createElementWithValidation @ react.development.js:1755
_renderStyleSelector @ static_dynamic_style_row.js:123
render @ static_dynamic_style_row.js:152
finishClassComponent @ react-dom.development.js:14535
updateClassComponent @ react-dom.development.js:14490
beginWork @ react-dom.development.js:15438
performUnitOfWork @ react-dom.development.js:19106
workLoop @ react-dom.development.js:19146
renderRoot @ react-dom.development.js:19229
performWorkOnRoot @ react-dom.development.js:20136
performWork @ react-dom.development.js:20048
performSyncWork @ react-dom.development.js:20022
interactiveUpdates$1 @ react-dom.development.js:20289
interactiveUpdates @ react-dom.development.js:2170
dispatchInteractiveEvent @ react-dom.development.js:4882

```


```
checkPropTypes.js:20 Warning: Failed prop type: The prop `color` is marked as required in `ColorRampSelect`, but its value is `null`.
    in ColorRampSelect (created by DynamicColorSelection)
    in DynamicColorSelection (created by StaticDynamicStyleRow)
    in div (created by EuiFormRow)
    in EuiFormRow (created by StaticDynamicStyleRow)
    in div (created by EuiFlexItem)
    in EuiFlexItem (created by StaticDynamicStyleRow)
    in div (created by EuiFlexGroup)
    in EuiFlexGroup (created by StaticDynamicStyleRow)
    in StaticDynamicStyleRow (created by VectorStyleColorEditor)
    in VectorStyleColorEditor (created by VectorStyleEditor)
    in VectorStyleEditor (created by StyleSettings)
    in div (created by EuiPanel)
    in EuiPanel (created by StyleSettings)
    in StyleSettings (created by Connect(StyleSettings))
    in Connect(StyleSettings) (created by LayerPanel)
    in div (created by LayerPanel)
    in div (created by LayerPanel)
    in div (created by EuiFlexGroup)
    in EuiFlexGroup (created by LayerPanel)
    in LayerPanel (created by Connect(LayerPanel))
    in Connect(LayerPanel) (created by GisMap)
    in div (created by EuiFlexItem)
    in EuiFlexItem (created by GisMap)
    in div (created by EuiFlexGroup)
    in EuiFlexGroup (created by GisMap)
    in GisMap (created by Connect(GisMap))
    in Connect(GisMap)
    in IntlProvider (created by I18nProvider)
    in I18nProvider
    in Provider
```